### PR TITLE
Enable multiple SparkSQL execution concurrently

### PR DESCRIPTION
**Pull request by minahlee**
This PR let user run multiple sql simultaneously.
But this feature is experimental, 
since it works fine only when there is one spark interpreter is bound to notes.

Screen shot for multiple sql running

### CODE REVIEW

File: spark/src/main/java/com/nflabs/zeppelin/spark/SparkSqlInterpreter.java [hunk 1]
```diff
@@ -282,7 +292,9 @@ public int getProgress(InterpreterContext context) {
 
   @Override
   public Scheduler getScheduler() {
```
Comment (Leemoonsoo) [inline spark/src/main/java/com/nflabs/zeppelin/spark/SparkSqlInterpreter.java]: Can you make `getScheduler()` conditionally return `getSparkInterpreter().getScheduler();` or `SchedulerFactory.singleton().createOrGetParallelScheduler(...)` depending on newly added property `concurrentSQL()` ?

**Comment by Leemoonsoo** [comment]
I tested this branch and it works really well.
The problem @minahlee mention is not oriented from this branch's feature. I created an issue #331 for the problem.
So I think it doesn't need to be experimental and it's good to be merged.

**Comment by swkimme** [comment]
One question.
With this setting, will a spark-sql paragraph run even if during a spark
paragraph is running?

**Comment by minahlee** [comment]
By default, Zeppelin is set to run the jobs in FAIR mode. Each pool gets an equal share of the cluster if there are multiple pools, but Zeppelin had only one scheduler pool(```default```) in fair scheduler. 
This PR adds one more scheduler pool named ```fair```.

Jobs run in FIFO inside ```default``` pool, while run parallel inside ```fair``` pool.

If you set zeppelin.spark.concurrentSQL 'true',
then spark paragraph will be assigned to ```default``` pool, while sparksql paragraph will be assigned to ```fair``` pool.

To make a long story short, yes, you can runs spark-sql paragraph even when spark paragraph is running.

**Comment by swkimme** [comment]
Thanks for great explaination!
It will be cooler if a proper docs is somewhere.
Anyway, LGTM

**Comment by Leemoonsoo** [comment]
Looks great! i'm merging it.

**Pull request merged**